File: java-junit5/src/test/java/org/codecop/Session2a_WordCounterTest.java
package org.codecop;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

/**
 * Session 2a: WordCounterTest - Basic assertions. <br />
 * @see "https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions"
 */
class Session2a_WordCounterTest {

    // TODO Add the proper assertions to complete the tests. 
    // The test name explains what needs to be asserted. 
    
    @Test
    void shouldCountNumberOfWords() {
        WordCounter counter = new WordCounter("Keep the bar green to keep the code clean.");
        // TODO Check that 9, counter.numberOfWords().
    }

    @Test
    void shouldVerifyContainmentOfWord() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that counter.containsWord("bar").
    }

    @Test
    void shouldVerifyNonContainmentOfWord() {
        WordCounter counter = new WordCounter("green hat");
        // TODO Check that counter.containsWord("red") is false.
    }

    @Test
    void shouldReturnNullForUnknownWordCount() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that counter.countOf("else") is null.
    }

    @Test
    void shouldReturnNotNullWordCountForExistingWord() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that counter.countOf("green") is not null.
    }

    @Test
    void shouldFindUniqueWords() {
        WordCounter counter = new WordCounter("green bar green hat");
        // TODO Check that new String[] { "bar", "green", "hat" }, counter.uniqueWords().
    }

    @Test
    void shouldReturnRatioOfWords() {
        WordCounter counter = new WordCounter("green bar green");
        // TODO Check that 0.33, counter.ratioOf("bar").
        // Note that floating point numbers need an accuracy delta, e.g. 0.01.
    }
}

File: csharp-xunit/WordCount.Tests/Session2a_WordCounterTest.cs
using Xunit;

namespace Org.Codecop.WordCount.Tests
{
    /// <summary>Session 2a: WordCounterTest - Basic assertions.</summary>
    /// <seealso>"https://xunit.github.io/docs/comparisons#assertions"</seealso>
    public class Session2a_WordCounterTest
    {

        // TODO Add the proper assertions to complete the tests.
        // The test name explains what needs to be asserted.

        [Fact]
        public void ShouldCountNumberOfWords()
        {
            var counter = new WordCounter("Keep the bar green to keep the code clean.");
            // TODO Check that 9 and counter.NumberOfWords() are equal.
        }

        [Fact]
        public void ShouldVerifyContainmentOfWord()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that counter.ContainsWord("bar") is true.
        }

        [Fact]
        public void ShouldVerifyNonContainmentOfWord()
        {
            var counter = new WordCounter("green hat");
            // TODO Check that counter.ContainsWord("red") is false.
        }

        [Fact]
        public void ShouldReturnNullForUnknownWordCount()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that counter.CountOf("else") is null.
        }

        [Fact]
        public void ShouldReturnNotNullWordCountForExistingWord()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that counter.CountOf("green") is not null.
        }

        [Fact]
        public void ShouldFindUniqueWords()
        {
            var counter = new WordCounter("green bar green hat");
            // TODO Check that new string[] { "bar" and "green", "hat" }, counter.UniqueWords() are equal.
        }

        [Fact]
        public void ShouldReturnRatioOfWords()
        {
            var counter = new WordCounter("green bar green");
            // TODO Check that 0.333 and counter.RatioOf("bar") are equal.
            // Note that floating point numbers need an accuracy precision, e.g. 3 digits.
        }
    }
}
